EventFilter/RawDataCollector/src/RawDataMapperByLabel.cc

```diff
@@ -0,0 +1,113 @@
+    RawDataMapperByLabel(const edm::ParameterSet& pset);
+    
+    ///Destructor
+    ~RawDataMapperByLabel() override;
+    
+    void produce(edm::Event & e, const edm::EventSetup& c) override; 
+    
+    static void fillDescriptions(edm::ConfigurationDescriptions &);     
+private:
+    
+    typedef std::vector<edm::InputTag>::const_iterator tag_iterator_t;
+    typedef std::vector<edm::EDGetTokenT<FEDRawDataCollection> >::const_iterator tok_iterator_t;
+
+    std::vector<edm::InputTag> inputTags_ ;
+    std::vector<edm::EDGetTokenT<FEDRawDataCollection> > inputTokens_;
+    
+    bool firstEvent_;
+    edm::InputTag filledCollectionName_;
+    edm::InputTag mainCollectionTag_ ;
+    
+    
+
+};
+
+
+RawDataMapperByLabel::RawDataMapperByLabel(const edm::ParameterSet& pset) {
+
+  inputTags_ = pset.getParameter<std::vector<InputTag> >("rawCollectionList");
+  mainCollectionTag_ = pset.getParameter<InputTag>("rainCollection");
+  firstEvent_= true;
+  filledCollectionName_= InputTag("");
+
+  inputTokens_.reserve(inputTags_.size());
+  for(auto const& inputTag: inputTags_) {
+    inputTokens_.push_back(consumes<FEDRawDataCollection>(inputTag));
+  }
+  produces<FEDRawDataCollection>();
+ 
+}
+
+RawDataMapperByLabel::~RawDataMapperByLabel(){
+
+}
+
+
+void RawDataMapperByLabel::produce(Event & e, const EventSetup& c){
+ 
+ bool alreadyACollectionFilled= false;
+ tag_iterator_t inputTag = inputTags_.begin();
+ for(tok_iterator_t inputTok = inputTokens_.begin(); inputTok != inputTokens_.end(); ++inputTok, ++inputTag  ) {
+   Handle<FEDRawDataCollection> input;
+   if (e.getByToken(*inputTok,input)){
+    	if(input.isValid()){
+    	    if(firstEvent_){  
+    	    	filledCollectionName_ = *inputTag; 
+    	    	alreadyACollectionFilled = true;
+                firstEvent_= false;  
+            }
+            
+    		if(alreadyACollectionFilled) throw cms::Exception("Unknown input type") << "Two input collections are present. Please make sure that the input dataset has only one FEDRawDataCollector collection filled";
+            if(!(filledCollectionName_==*inputTag)) throw cms::Exception("Unknown input type") << "The filled collection has changed!";
+            
+            if(!(mainCollectionTag_==filledCollectionName_)) e.put(std::make_unique<FEDRawDataCollection>(*input.product()));
+                      
+        }
+    }
+  }
+}
+
+void RawDataMapperByLabel::fillDescriptions(edm::ConfigurationDescriptions & descriptions) {
+    edm::ParameterSetDescription desc;
+    
+    std::vector<edm::InputTag> tmp_itvect;
+    tmp_itvect.push_back( edm::InputTag("rawDataCollector"));
+    tmp_itvect.push_back( edm::InputTag("rawDataRepacker"));
+    tmp_itvect.push_back( edm::InputTag("rawDataReducedFormat"));
+    
+    desc.add<std::vector<edm::InputTag>>("rawCollectionList",  tmp_itvect);
+    desc.add<edm::InputTag>("mainCollection", edm::InputTag("rawDataCollector"));
+    
+    descriptions.add("rawDataCollector", desc);
```

Comment:
this will create a local file  EventFilter.RawDataCollector.rawDataCollector_cfi
such file already exists in the source tree
this is a problem

Reply:
I commented out the tmp_itvect filling. If I understood correctly from https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

The module should then simply check that as a parameters are passed a cms.VInputTag and cms.InputTag. If the VInputTag is empty, the module will simply not run. 

Other alternative would be to keep only the declaration of the fillDescription without implementation. Please advise

Reply:
> I commented out the tmp_itvect filling

I'm not sure I follow this response.
My comment was about the ```descriptions.add("rawDataCollector", desc)``` line

Reply:
@slava77 do you think that I should comment that line? not sure I understand the reason but I can do it. module also renamed "rawDataMapper"

Reply:
rename is good enough. However, a more appropriate default is ```rawDataMapperByLabel```
(a lower case of the module class name). After that change, it will be time to remove the explicit cfi file introduced in this PR, which is now a duplicate of what the framework will build for you already (the ```descriptions.add("rawDataCollector"``` leads to a creation of rawDataCollector_cfi.py file in $CMSSW_BASE/cfi\*python/